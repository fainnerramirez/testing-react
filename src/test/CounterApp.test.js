import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en el componente <CounterApp />", () => {
  test("se debe mostrar correctamente <CounterApp />", () => {
    const view = shallow(<CounterApp />);

    expect(view).toMatchSnapshot();
  });

  test("debe de mostrar el valor por defecto de 100", () => {
    const valor = "100";

    const view = shallow(<CounterApp value={100} />);

    const textoParrafo = view.find("h1").text().trim();

    console.log(textoParrafo);

    expect(textoParrafo).toBe(valor);
  });
});
