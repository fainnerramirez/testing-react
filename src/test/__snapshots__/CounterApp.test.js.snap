// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CounterApp /> se debe mostrar correctamente <CounterApp /> 1`] = `
<div
  className="App"
>
  <h1>
    100
  </h1>
  <button
    onClick={[Function]}
  >
    increment
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
